react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      val: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ val: e.target.value });
+  }
+
+  handleClick(e) {
+    e.preventDefault();
+  }
+
+  render() {
+    return (
+      <form>
+        <label>
+          Email:
+          <input type='text' name='Email' onChange={this.handleChange} />
```

Comment:
Here you need to have a `value` attribute set with your value for the input in state, in this case `this.state.val`

Reply:
Updated